### PR TITLE
Implementing SendMessageOneway

### PR DESCRIPTION
**Pull request by wenfengwang**
1. Implementing `SendMessageOneway` method
2. other fix: 
    - Aligning fields in `String()` of config
    - Optimizing `examples/producer.go`
    - Optimizing definition of `MessageModel`

### CODE REVIEW

File: core/push_consumer.go [hunk 1]
```diff
@@ -169,10 +167,20 @@ func newPushConsumer(config *PushConsumerConfig) (PushConsumer, error) {
...
+		switch config.Model {
+		case BroadCasting:
+			mode = C.BROADCASTING
+		case Clustering:
+			mode = C.CLUSTERING
+		}
+		code = int(C.SetPushConsumerMessageModel(cconsumer, mode))
+
+		if code != 0 {
+			return nil, errors.New(fmt.Sprintf("PushConsumer Set ConsumerMessageModel error, code is: %d", int(code)))
```
Comment (zjykzk) [inline core/push_consumer.go]: `fmt.Errorf` is simpler

File: core/utils.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package rocketmq
+
+import "fmt"
+
+func strJoin(str, key string, value interface{}) string {
```
Comment (zjykzk) [inline core/utils.go]: `fmt.Stringer` interface is better than `interface{}`

Comment (wenfengwang) [reply core/utils.go]: value passed in maybe is `int` ~ : )

Comment (zjykzk) [reply core/utils.go]: GOOD

File: core/push_consumer.go (hunk 1 above)
Comment (wenfengwang) [reply core/push_consumer.go]: fixed

File: core/api.go [hunk 3]
```diff
@@ -32,6 +32,22 @@ type clientConfig struct {
 	LogC             *LogConfig
 }
 
+func (config *clientConfig) string() string {
```
Comment (zjykzk) [inline core/api.go]: `string()` -> `String()`, it's typo?

Comment (wenfengwang) [reply core/api.go]: no typo, the method is a helper method for sub-struct(`ProducerConfig\PushConsumerConfig`) `String()`

**Pull request merged**